packages/charts/src/Charts/SeverityLine/SeverityLine.js

```diff
@@ -0,0 +1,71 @@
+import './severity-line.scss';
+
+import React, { useEffect, useState } from 'react';
+
+import PropTypes from 'prop-types';
+import { Tooltip } from '@patternfly/react-core';
+
+const chartMapper = (width) => ({
+    1: '1',
+    2: (width / 3),
+    3: (width * (2 / 3)),
+    4: (width - 1)
+});
+
+const SeverityLine = ({ title, value, tooltipMessage, config, chartProps }) => {
+
+    const [ severity, setSeverity ] = useState();
+    const { width, height } = config;
+    const points = [
+        { className: 'horizontal', points: `${1}, ${height / 2} ${width - 1}, ${height / 2}` },
+        { className: 'vertical', points: `${width - 1}, ${0} ${width - 1}, ${height}` },
+        { className: 'vertical', points: `${width * (2 / 3)}, ${0} ${width * (2 / 3)}, ${height}` },
+        { className: 'vertical', points: `${width / 3}, ${0} ${width / 3}, ${height}` },
+        { className: 'vertical', points: `${1}, ${0} ${1}, ${height}` },
+        { className: `dataline-${value}`, points: `${1}, ${height / 2} ${severity}, ${height / 2}` }
+    ];
+
+    useEffect(() => {
+        setSeverity(chartMapper(width)?.[value]);
+        //eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [ value ]);
+
+    return <div className='ins-c-severity-line' { ...chartProps }>
+        <div className='ins-l-title'>
+            {title.length > 14 ? <Tooltip content={title}>
+                <span> {`${title.substring(0, 14)}...`} </span>
+            </Tooltip> : <span> {title} </span>}
+        </div>
+        {severity && <div className='ins-l-chart'>
```

Comment:
I think you don't have to use state `severity`. Just compute the value here.

Reply:
`severity` is used on multiple places, I think it's fine to keep as state variable. Otherwise the variable would have been calculated on each re-render.

Reply:
Is severity ever changed? If not  `useRef` is a better way to persist variables like that than `useState`.

Reply:
Yeah, it can be computed in a variable and used multiple times. And it's not so complex computation and I think the only reason to re-render this component will be to update the value, so in the end it will be basically the same, just with no need to use expensive hook. (And because of the useEffect, the component will be rendered twice on the initial render.) We can use memo to optimize it further.